ingestors/pypi_xml_rpc.go

```diff
@@ -65,10 +73,8 @@ func (ingestor *PyPiXmlRpc) Ingest() []data.PackageVersion {
 		}
 	}
 
-	if len(results) > 0 {
-		if _, err := setBookmarkTime(ingestor, data.MaxCreatedAt(results)); err != nil {
-			log.WithFields(log.Fields{"ingestor": ingestor.Name()}).Fatal(err)
-		}
+	if _, err := setBookmarkTime(ingestor, data.MaxCreatedAt(results)); err != nil {
+		log.WithFields(log.Fields{"ingestor": ingestor.Name()}).Fatal(err)
```

Comment:
I don't see a reason why we only set the bookmark if there are results: if there are legitimately no results, it should be fine to update the timestamp so we skip this current page next time.

Reply:
Because you'd be setting the timestamp to now() which might not be synchronized to the feed, you only know that you're "current" if you get a timestamp from the feed.

Reply:
In this case I think you're gonna set the bookmark to the "zero value" of time.Time which is 0001-01-01 00:00:00 +0000 UTC. I don't think you wanna do that.

Reply:
Oh snap, thanks for noticing that! Pushing a fix in a sec.

> you only know that you're "current" if you get a timestamp from the feed.

I don't think there's a reliable way to grab a timestamp from the feed if we've thrown it out anyway (until we have time to spelunk the go codebase and figure out how to recover from those).